compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirNonLocalFunctionChecker.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.analysis.checkers.extended.report
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.declarations.FirSimpleFunction
+import org.jetbrains.kotlin.fir.declarations.hasBody
+
+internal interface FirNonLocalFunctionChecker {
```

Comment:
I'd remove this interface (as well as `FirNonLocalPropertyChecker`). With a lot of `FirSomethingChecker` walking around this interfaces look strange. I'd make functions from them top-level utility functions, with `internal` visibility and better names, like `checkExpectFunctionHasNoBody`.

Reply:
The purpose of this PR is to get feedback about the overall structure, so thanks for the suggestion. Removed *NonLocal* interfaces.

Regarding the name of that particular check, yeah, I planned to add additional check, private expected declaration, into `checkExpectFunction`. But, maybe it's better to be specific. I added private expected declaration check at 4th commit, and made each util to check visibility/modality explicitly as well.